djangoguestbook/guestbook/views.py

```diff
@@ -1,45 +1,45 @@
-import urllib
-import logging
-from django.http import HttpResponseRedirect
-from django.shortcuts import render
+from django.views.generic.base import TemplateView
+from guestbook.models import Greeting, get_guestbook_key, DEFAULT_GUESTBOOK_NAME
 from google.appengine.api import users
+from django.http import HttpResponseRedirect
 from google.appengine.api import memcache
-from guestbook.models import Greeting, get_guestbook_key, DEFAULT_GUESTBOOK_NAME
-
-def main_page(request):
-    guestbook_name = request.GET.get('guestbook_name', DEFAULT_GUESTBOOK_NAME)
-    greetings = get_greetings(guestbook_name)
-    stats = memcache.get_stats()
-    if users.get_current_user():
-        url = users.create_logout_url(request.get_full_path())
-        url_linktext = 'Logout'
-    else:
-        url = users.create_login_url(request.get_full_path())
-        url_linktext = 'Login'
-    template_values = {
-        'greetings': greetings,
-        'guestbook_name': guestbook_name,
-        'url': url,
-        'url_linktext': url_linktext,
-        'statshits': stats['hits'],
-        'statsmisses': stats['misses'],
-    }
-    return render(request, 'guestbook/main_page.html', template_values)
-
-def sign_post(request):
-    if request.method == 'POST':
-        guestbook_name = request.POST.get('guestbook_name')
-        greeting = Greeting(parent=get_guestbook_key(guestbook_name))
+import urllib
+import logging
+class GreetingView(TemplateView):
+    template_name = "guestbook/main_page.html"
+    def get(self, request, *args, **kwargs):
```

Comment:
xài TemplateView thì không cần định nghĩa lại get method
hãy vào TemplateView & parent of TemplateView
- xem các method đã có sẵn
- định nghĩa lại method cần (Vd: get_context_data )
 - trong get_context_data( )  có thể lấy GET param thông qua self.request.GET